dask/array/core.py

```diff
@@ -1221,12 +1224,7 @@ def __getitem__(self, index):
 
         dsk, chunks = slice_array(out, self.name, self.chunks, index)
 
-        if len(dsk) < self.npartitions / 2:  # significant reduction in graph
-            needed = core.get_dependencies(self.dask, task=list(dsk.values()))
```

Comment:
Yes.  This was originally designed to reduce the size of the graph early on in computations.  However it is sometimes expensive.  Here is an informative use case:

```python
L = [x[0, :] for i in [...]
```

Before culling this would create N copies of the underlying graph.  This was very bad.  In master this calls cull N times, which is sort of bad but not as bad as before.  Now we don't need to worry about these copies, so we don't cull at all.  This is good.